src/common/snippets/src/op/subgraph.cpp

```diff
@@ -369,35 +368,30 @@ void snippets::op::Subgraph::align_element_types(const BlockedShapeVector& outpu
             const auto convert = std::make_shared<ngraph::snippets::op::ConvertSaturation>(
                 body_results[i]->get_input_node_shared_ptr(0), needed_out_type);
             body_results[i]->set_argument(0, convert);
+            body_results[i]->validate_and_infer_types();
         }
     }
 
     // We should change existing element type to original for Parameters if needed
-    const auto& body_parameters = body_ptr()->get_parameters();
+    const auto& parameters = body_ptr()->get_parameters();
     for (size_t i = 0; i < inputShapes.size(); ++i) {
         const auto needed_in_type = std::get<2>(inputShapes[i]);
-        if (body_parameters[i]->get_element_type() != needed_in_type) {
-            body_parameters[i]->set_element_type(needed_in_type);
-            config.m_is_needed_to_align_precision = true;
-        }
-    }
+        if (parameters[i]->get_element_type() != needed_in_type) {
+            const auto output = parameters[i]->output(0);
+            for (const auto input : output.get_target_inputs()) {
```

Comment:
As I got it, each Consumer will have own `Convert` on input. But it's not optimal for performance if there will be the same several `Convert` nodes. I think it's written so for more comfortable precision propagation. But if we imagine the below situation:
![image](https://user-images.githubusercontent.com/43129309/220051859-35a0f2c9-b6cd-4dbd-9b73-619fb05a6322.png)
we will inference each `Convert`.

There are two ways. First, we insert one common `Convert` node and update `PrecisionPropagation` if needed. Second, we add pass to collapse these `Convert` nodes to one after `PrecisionPropagation` if possible. But I'd like to avoid new passes to not increase model compilation time. What do you think? Maybe I missed something

Reply:
good note, thanks. Everyting was done in the same pass with some optimization. No additional passes.
Test is WIP.